dags/utils/dataproc.py

```diff
@@ -0,0 +1,150 @@
+from airflow import models
+from airflow.contrib.hooks.gcp_api_base_hook import GoogleCloudBaseHook
+from airflow.contrib.operators.dataproc_operator import DataprocClusterDeleteOperator, DataProcSparkOperator
+from airflow.exceptions import AirflowException
+from airflow.utils import trigger_rule
+
+# Our own dataproc operator used to install optional components and component gateway
+from operators.moz_dataproc_operator import DataprocClusterCreateOperator
+
+
+def _format_envvar(env=None):
+    # use a default value if an environment dictionary isn't supplied
+    return ' '.join(['{}={}'.format(k, v) for k, v in (env or {}).items()])
+
+def moz_dataproc_runner(parent_dag_name=None,
+                        dag_name='run_script_on_dataproc',
+                        default_args=None,
+                        cluster_name=None,
+                        num_workers=2,
+                        image_version='1.4',
+                        zone='us-central1-b',
+                        idle_delete_ttl='14400',
+                        auto_delete_ttl='28800',
+                        master_machine_type='n1-standard-8',
+                        worker_machine_type='n1-standard-4',
+                        num_preemptible_workers=0,
+                        service_account=None,
+                        job_name=None,
+                        owner=None,
+                        uri=None,
+                        env=None,
+                        arguments='',
+                        artifact_bucket='moz-fx-data-wtmo-artifacts',
+                        output_bucket='moz-fx-data-prod-airflow-output',
+                        gcp_conn_id='google_cloud_airflow_dataproc'):
+
+    """
+    This executes a Spark Job on Dataproc via a script Uri. This will execute 
```

Comment:
The wording here is difficult to follow, in my opinion.

Reply:
changed